packages/blade/src/components/Table/TableEditableCell.web.tsx

```diff
@@ -0,0 +1,174 @@
+          <Box margin={getEditableInputMargin({ rowDensity })} width="100%">
+            <BaseInput
+              isTableInputCell={rowDensityToIsTableInputCellMapping[rowDensity]}
+              validationState={validationState}
+              id="table-editable-cell-input"
+              type="text"
+              size={tableEditableCellRowDensityToInputSizeMap[rowDensity]}
+              trailingIcon={validationStateToInputTrailingIconMap[validationState]}
+              accessibilityLabel={accessibilityLabel}
+              autoCapitalize={autoCapitalize}
+              autoCompleteSuggestionType={autoCompleteSuggestionType}
+              // eslint-disable-next-line jsx-a11y/no-autofocus
+              autoFocus={autoFocus}
+              defaultValue={defaultValue}
+              isDisabled={isDisabled}
+              isRequired={isRequired}
+              keyboardReturnKeyType={keyboardReturnKeyType}
+              leadingIcon={leadingIcon}
+              maxCharacters={maxCharacters}
+              name={name}
+              onBlur={onBlur}
+              onChange={onChange}
+              onClick={onClick}
+              onFocus={onFocus}
+              onSubmit={castWebType(onSubmit)}
+              placeholder={placeholder}
+              prefix={prefix}
+              suffix={suffix}
+              value={value}
+              testID={testID}
+              trailingButton={trailingButton}
+              errorText={errorText}
+              successText={successText}
+              showHintsAsTooltip={true}
+            />
+          </Box>
+        </CellWrapper>
+      </BaseBox>
+    </StyledEditableCell>
+  );
+};
+
+const TableEditableDropdownCell = (
+  dropdownProps: TableEditableDropdownCellProps,
+): React.ReactElement => {
+  const { rowDensity, showStripedRows, backgroundColor } = useTableContext();
+
+  return (
+    <TableEditableCellContext.Provider value={{ isInsideTableEditableCell: true }}>
+      <StyledEditableCell
+        role="cell"
+        $backgroundColor={backgroundColor}
+        rowDensity={rowDensity}
+        {...metaAttribute({ name: MetaConstants.TableCell })}
+      >
+        <BaseBox
+          className="cell-wrapper-base"
+          display="flex"
+          alignItems="center"
+          height="100%"
+          width="100%"
+        >
+          <CellWrapper
+            className="cell-wrapper"
+            rowDensity={rowDensity}
+            showStripedRows={showStripedRows}
+            display="flex"
+            alignItems="center"
+            flex={1}
+            hasPadding={false}
+          >
+            <Dropdown _width="100%" margin="spacing.2" {...dropdownProps} />
+          </CellWrapper>
+        </BaseBox>
+      </StyledEditableCell>
+    </TableEditableCellContext.Provider>
+  );
+};
+
+const TableEditableCell = assignWithoutSideEffects(_TableEditableCell, {
+  componentId: ComponentIds.TableEditableCell,
+});
+
+export { TableEditableCell, TableEditableDropdownCell };
```

Comment:
Since we are keeping them in the same place, won't this cause a problem with bundle size? If I am not using dropdown cell but only need editable cell, this would end up importing dependencies for dropdown as well, right?

Reply:
Nope the webpack will follow the imports and usage so it won't import things that are not used in that component